File: app/src/main/java/com/vpaliy/mediaplayer/media/playback/Playback.java
package com.vpaliy.mediaplayer.media.playback;

/* The base interface for any playback implementation */
public interface Playback  {

    void play(String mediaUrl);
    void pause();
    void stop();
    void seekTo(int position);
    void setCallback(Callback callback);
    boolean isPlaying();
    long getPosition();

    /* This interface is used to notify others about changes in the player*/
    interface Callback {
        void onCompletetion();
        void onPause();
        void onPlay();
        void onStop();
        void onError();
    }
}

File: app/src/main/java/com/vpaliy/mediaplayer/domain/playback/Playback.kt
package com.vpaliy.mediaplayer.domain.playback

interface Playback {

    fun play(streamUrl: String?)
    fun pause()
    fun stop()
    fun seekTo(position: Int)
    fun invalidateCurrent()
    fun isPlaying(): Boolean
    fun position(): Long
    fun assignCallback(callback:Callback)

    interface Callback {
        fun onCompletetion()
        fun onPause()
        fun onPlay()
        fun onStop()
        fun onError()
    }
}